libs/core/algorithms/include/hpx/parallel/unseq/simd_helpers.hpp

```diff
@@ -0,0 +1,153 @@
+//  Copyright (c) 2022 A Kishore Kumar
+//  Copyright (c) 2023 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/functional/detail/invoke.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/parallel/unseq/reduce_helpers.hpp>
+#include <hpx/type_support/construct_at.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace hpx::parallel::util {
+    // borrowed from Intel Implementation found in the GCC library
+    template <class Iter, class DifferenceTy, class F>
+    Iter unseq_first_n(Iter first, DifferenceTy beginIndex,
+        DifferenceTy endIndex, F&& f) noexcept
+    {
+#if HPX_EARLYEXIT_PRESENT
+        DifferenceTy __i = beginIndex;
+
+        // clang-format off
+        HPX_PRAGMA_VECTOR_UNALIGNED HPX_PRAGMA_SIMD_EARLYEXIT
+        for (; __i < endIndex; ++__i)
+        {
+            if (f(first, __i))
+            {
+                break;
+            }
+        }
+        // clang-format on
+
+        return first + __i;
```

Comment:
This code was taken from libstdc++, was it? Please be aware that this library is GPL licensed and we're not able to add it to our code base.

Reply:
ok, will ideate on a different implementation.